src/backend/functions/orchestrator/lib/index.js

```diff
@@ -102,35 +98,103 @@ function createDeleteParams({mediaUrl, tasksTableName}) {
    };
 }
 
-// terminated :: ({k:v} -> Promise {k: v}) -> {k: v} -> String -> Promise {k:v}
+// terminated :: ({k:v} -> Promise {k: v}) -> {k: v} -> {k: v} -> Promise {k:v}
 const terminated = R.curry(deleteItem =>
    handle(o(deleteItem, createDeleteParams), ({MediaUrl: mediaUrl}) => ({mediaUrl}))
 );
 
-function error(ecs, ddb) {}
+const errorUpdate = R.curry((updateItem, {tasksTableName, mediaUrl, taskArn}) => {
+   const params = {
+      TableName: tasksTableName,
+      Key: {
+         MediaUrl: mediaUrl
+      },
+      UpdateExpression: 'ADD Retries :val SET TaskStatus = :status, TaskArn = :task',
+      ExpressionAttributeValues: {
+         ':val': 1,
+         ':status':  'WAITING',
+         ':task': taskArn
+      },
+      ReturnValues: 'ALL_NEW'
+   };
+
+   return updateItem(params)
+       .then(({Attributes}) => ({...Attributes, tasksTableName}))
+});
 
+const unrecoverableError = R.curry((updateItem, {tasksTableName, MediaUrl}) => {
```

Comment:
just for consistency, maybe it'd be better to rename this function? as your handler functions all match the status currently except this one i.e. ERROR -> error, WAITING -> waiting, TERMINATING -> terminating etc.

Reply:
I just converted `UNRECOVERABLE_ERROR` to camelCase like it I do for the environment variables although I think I need to make this more consistent (see my response to renaming destructuring below).